documentation/docs/developers/strategies/perform-trade.md

```diff
@@ -0,0 +1,42 @@
+# Perform Trade
+
+## Architecture
+
+In the Perform Trade strategy, Hummingbot places a single order that is specified by the user. Unlike Simple Trade, orders aren’t cancelled after a certain amount of time and are only placed once at the start of the strategy.
+
+Here's a high level view of the logic flow inside the built-in perform trade strategy.
+
+![Figure 1: Perform trade strategy flow chart](/assets/img/perform-trade-flowchart.svg)
+
+The flow of the strategy is as follows
+
+1. Get user input
+2. Sanity Checks
+3. Place Order
+
+
+## Sanity Checks & Validation
+
+Before the strategy attempts to perform any trades it checks the following to ensure it is safe to perform the trade:
+
+1. Are both markets ready and connected?
+2. Does the user have enough balance? 
+    1. Additionally, this check triggers a warning and logs the warning. If the user does not have the requisite balance to perform the trade, the interface will display the warning.
+    2. To achieve this check, the Perform Trade strategy gets the user’s available balance by using the c_has_enough_balance in perform_trade.pyx.
+
+## Managing Order State
+Order state is tracked using the quantity, price, type, id, and other attributes. 
+The OrderTracker class includes functions monitoring when order tracking begins, noticing and tracking cancels, and when tracking ends. When an order is completed an order id is returned and the specifics concerning the order can be determined by using the OrderTracker class.
+
+The order state, specifically, whether the order is partially or completely filled is managed by events in the MarketBase class. Specifically, the logic regarding order creation, cancelation, filling, and completion can be found in the following functions in hummingbot/market/market_base.pyx: buy, sell, cancel, get_order_price_quantum (gets the required or allowed order price), and get_order_size_quantum(gets the required or allowed order size).
+
+The order tracking logic can be found in the c_start_tracking_limit_order, c_stop_tracking_limit_order, c_has_in_flight_cancel and c_check_and_track_cancel function inside hummingbot/strategy/order_tracker.pyx.
```

Comment:
Please quote function names as code in markdown. Also, function names should come with "()" to indicate they're functions.

So instead of "c_has_in_flight_cancel", it should be `c_has_in_flight_cancel()`.

Another thing is the links for source code files. For example, instead of "perform_trade.pyx", it should be [perform_trade.pyx](https://github.com/CoinAlpha/hummingbot/blob/development/hummingbot/strategy/dev_2_perform_trade/dev_2_perform_trade.pyx).

You can use Macdown to edit the mark down files to make formatting easier.

Reply:
Quoted the functions as code and added links to the files mentioned!